server/src/infra/repositories/search.repository.ts

```diff
@@ -249,4 +251,86 @@ export class SearchRepository implements ISearchRepository {
 
     return runtimeConfig;
   }
+
+  async searchHybrid(
+    pagination: SearchPaginationOptions,
+    { embedding, userIds, ...options }: HybridSearchOptions,
+  ): Paginated<AssetEntity> {
+    let results: PaginationResult<AssetEntity> = { items: [], hasNextPage: false };
+
+    await this.assetRepository.manager.transaction(async (manager) => {
+      let builder = manager.createQueryBuilder(AssetEntity, 'asset');
+
+      builder.where('asset.ownerId IN (:...userIds )', { userIds }).leftJoinAndSelect('asset.exifInfo', 'exif');
+
+      if (embedding && embedding.length > 0) {
+        builder
+          .innerJoin('asset.smartSearch', 'search')
+          .orderBy('search.embedding <=> :embedding')
+          .setParameters({ embedding: asVector(embedding) });
+      }
+
+      if (options.state) {
+        builder.andWhere('exif.state = :state', { state: options.state });
+      }
+
+      if (options.country) {
+        builder.andWhere('exif.country = :country', { country: options.country });
+      }
+
+      if (options.city) {
+        builder.andWhere('exif.city = :city', { city: options.city });
+      }
+
+      if (options.make) {
+        builder.andWhere('exif.make = :make', { make: options.make });
+      }
+
+      if (options.model) {
+        builder.andWhere('exif.model = :model', { model: options.model });
+      }
+
+      if (options.createdAfter) {
+        builder.andWhere('exif.dateTimeOriginal >= :createdAfter::date', { createdAfter: options.createdAfter });
+      }
+
+      if (options.createdBefore) {
+        builder.andWhere('exif.dateTimeOriginal <= :createdBefore::date', { createdBefore: options.createdBefore });
+      }
+
+      if (options.isArchived) {
+        builder.andWhere('asset.isArchived = :isArchived', { isArchived: options.isArchived });
+      }
+
+      if (options.isFavorite) {
+        builder.andWhere('asset.isFavorite = :isFavorite', { isFavorite: options.isFavorite });
+      }
```

Comment:
All of this duplicates what's already handled in `searchSmart`. The only difference between what that endpoint supports and the hybrid search I mentioned is that hybrid search also does full-text search on certain fields like the description. But filters like these don't need a separate hybrid search endpoint.